### PR TITLE
feat(cli-lib): experimental support for programatic CLI api

### PR DESCRIPTION
**Pull request by mrgrain**
----

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing guide?](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md)

### Adding new Unconventional Dependencies:

* [ ] This PR adds new unconventional dependencies following the process described [here](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md/#adding-new-unconventional-dependencies)

### New Features

* [ ] Have you added the new feature to an [integration test](https://github.com/aws/aws-cdk/blob/main/INTEGRATION_TESTS.md)?
	* [ ] Did you use `yarn integ` to deploy the infrastructure and generate the snapshot (i.e. `yarn integ` without `--dry-run`)?

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/aws-cdk-api/package.json [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+  "name": "aws-cdk-api",
+  "description": "AWS CDK Programmatic API",
+  "version": "0.0.0",
+  "main": "lib/index.js",
+  "types": "lib/index.d.ts",
+  "jsii": {
+    "outdir": "dist",
+    "targets": {
+      "java": {
+        "package": "software.amazon.awscdk.api",
```
Comment (rix0rrr) [inline packages/aws-cdk-api/package.json:11]: That's a bold name! Not sure how I feel about it 😬

File: packages/aws-cdk-api/lib/api.ts [hunk 2]
```diff
@@ -0,0 +1,226 @@
+import * as core from '@aws-cdk/core';
+import { exec as runCli } from 'aws-cdk/lib';
```
Comment (rix0rrr) [inline packages/aws-cdk-api/lib/api.ts:2]: I trust that this works in the working copy... I don't fully trust that this works after publishing to NPM, based on trickery and #$%*ery we do to the CLI build before we publish.

Does it?

**Review by rix0rrr — COMMENTED** [review]
Should we change the package name to something more along the lines of `cdk-cli-lib` ?

(not strictly that, but something in that vein?)

Ultimately, I'd think the goal would be for all code to live here and the dependency arrow to be reversed, right?

File: packages/aws-cdk-api/package.json (hunk 1 above)
Comment (mrgrain) [reply packages/aws-cdk-api/package.json:11]: Agreed. The linter forced that name based on the package name. But we might change the package name.

**Comment by mrgrain** [comment]
> Should we change the package name to something more along the lines of `cdk-cli-lib` 
> (not strictly that, but something in that vein?)

Yes, I wanted to ask for feedback here. Do we have rules for when we use the `@aws-cdk` prefix and when not?
I could see `@aws-cdk/cli-lib`

> Ultimately, I'd think the goal would be for all code to live here and the dependency arrow to be reversed, right?
That's 100% right.

File: packages/aws-cdk-api/lib/api.ts (hunk 2 above)
Comment (mrgrain) [reply packages/aws-cdk-api/lib/api.ts:2]: Yes it does work. This minimal example can be run in a shell:
```sh
npm install aws-cdk 
node -e "const { cli } = require('aws-cdk/lib'); cli()"
```

`cli` is the existing export. I have no reason to believe `exec` would behave any different. However it's tricky to proof with the current build system.

Maybe doing the code swap first would be better?

**Comment by mrgrain** [comment]
Closed in favor of https://github.com/aws/aws-cdk/pull/22836

**Pull request closed without merging**